plugin/defs.go

```diff
@@ -42,6 +44,30 @@ func WithSwarmService(id string) CreateOpt {
 	}
 }
 
+// WithEnv is a CreateOpt that passes the user-provided environment variables
```

Comment:
Can you update this to mention that it de-dups and only appends valid key=value pairs?

Reply:
Does this work better?
https://github.com/moby/moby/blob/8908c317daa6799744fbc0588e2a125359f8d84b/plugin/defs.go#L47-L50

Reply:
👍

Reply:
Hi @cpuguy83,

With this one resolved, is there anything else missing? I was hoping this could make it into 19.03 as it helps the use case of installing a plugin as a Swarm service.